lib/src/UrlEncoder.cpp

```diff
@@ -292,10 +291,9 @@ void UrlEncoder::AddTemplate(const std::string& new_template, const bool overwri
     try
     {
         // Get the pen group value
-        std::string_view match;
-        match = matches[1].str();
-        std::uint16_t base = match.starts_with("0x") ? 16 : match.starts_with("0b") ? 2 : 10;
-        pen_value = std::stoul(match.data(), nullptr, base);
+        const std::ssub_match match = matches[1];
+        std::uint16_t base = match.str().starts_with("0x") ? 16 : match.str().starts_with("0b") ? 2 : 10;
+        pen_value = std::stoul(match.str(), nullptr, base);
```

Comment:
In the case where the base is 16 we could probably use `quicr::hex_to_uint` which is constexpr, and could be faster. But that's really specific.